_rules/text-node-contrast-afw4f7.md

```diff
@@ -0,0 +1,345 @@
+---
+id: afw4f7
+name: Text node has minimal contrast
+rule_type: atomic
+
+description: |
+  This rule checks that every text node has minimal color contrast with its background
+
+accessibility_requirements:
+  wcag20:1.4.3: # Contrast (Minimum)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+  - Language
+
+authors:
+  - Brian Bors
+  - Kasper Isager
+  - Wilco Fiers
+---
+
+## Applicability
+
+Any [text node](https://dom.spec.whatwg.org/#text) that is [visible](#visible) and a [child](https://dom.spec.whatwg.org/#concept-tree-child) (in the [flat tree](https://drafts.csswg.org/css-scoping/#flat-tree)) of an HTML element, except if the [text node](https://dom.spec.whatwg.org/#text) is a [descendant](https://dom.spec.whatwg.org/#concept-shadow-including-descendant) of an element that:
```

Comment:
https://drafts.csswg.org/css-scoping/#flat-tree is from an Editors draft but is my understanding that there is no other definition available.

Reply:
This is correct.